SAMLPortal/Models/GlobalSettings.cs

```diff
@@ -20,29 +22,41 @@
 		/// </summary>
 		public static void InitSettingsFromEnvironment()
 		{
-			DotNetEnv.Env.Load("./.env");
-			IDictionary environment = Environment.GetEnvironmentVariables();
+			UpdateFromEnvironment();
 
-			if (Environment.GetEnvironmentVariable("SP_FORCE_ENV_REPOPUL") == "1")
+			string configurationPath = GlobalSettings.Get("CONFIG_PATH");
+			if (configurationPath != null)
 			{
-				// Re-fill databse with environment configuration
-				foreach (var key in environment.Keys)
+				if (Directory.Exists(configurationPath))
 				{
-					if (key.ToString().StartsWith("SP_") && !key.ToString().StartsWith("SP_MYSQL"))
+					// Ensure trailing slash is present
+					configurationPath = configurationPath.TrimEnd('/') + '/';
+
+					var fileConfigPath = configurationPath + "config.env";
+					if (!File.Exists(fileConfigPath))
 					{
-						string shortenedKey = key.ToString().Remove(0, 3);
-						_appSettings.Add(shortenedKey, environment[key].ToString());
+						using (FileStream fs = File.Create(fileConfigPath)) { }
+						Helpers.WriteEnvVariableToFile(fileConfigPath, "SP_CONFIG_SETUPASSISTANT_STEP", "0");
 					}
-				}
 
-				UpdateDatabaseSettings();
+					DotNetEnv.Env.Load(fileConfigPath);
+					Environment.SetEnvironmentVariable("SP_CONFIG_FILE", fileConfigPath);
+					UpdateFromEnvironment();
+
+					if (Environment.GetEnvironmentVariable("SP_MYSQL_USER") != null)
+					{
+						UpdateFromDatabase();
+					}
+				}
+				else
+				{
+					throw new Exception("The directory " + configurationPath + " does not exists. Please create it first.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)